lib/challenge.js

```diff
@@ -58,7 +75,21 @@ var some_pet;
 //    the values of `name`, `species`, `breed`, and `noise`. The constructor function should also give each pet a method called `makeNoise`,
 //    which behaves as described in Q3.
 
-var Pet;
+var Pet = function (name, species, breed, noise) {
+  this.name = name;
+  this.species = species;
+  this.breed = breed;
+  this.noise = noise;
+}
+
+var Pet = new Pet ("Neptune", "feline", "domestic shorthair", "meow");
+
+makeNoise : function (){
+  for (i = 0; i < some_pet.name.length; i++)
+    if (noise === this.Pet[i].noise) {
+
+//I'M STUCK!!!!
+}
```

Comment:
When you're using a Constructor function, methods for each 'instance' (in this case, a specific 'pet') get placed on the Constructor function's `prototype` property.

Here's how that might look. 

```javascript
var Pet = function(name, species, breed, noise) {
  ...
}
Pet.prototype.makeNoise = function(){
  console.log(this.noise);
};

var jake = new Pet("Jake", "dog", "burmese stretchy dog", "Makin' pancakes, makin' bacon pancakes.");
jake.makeNoise();
```